### PR TITLE
br: add batch ddl post restore for auto id

### LINKED ISSUE #32006: BR: incremental batch ddl restore cause table could not execute insert values
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. create database and table
`CREATE DATABASE IF NOT EXISTS db;`
`CREATE TABLE IF NOT EXISTS db.t(c1 INT);`
`INSERT INTO db.t(c1) VALUES (1);`
2. backup the table
`br --pd $PD_ADDR backup table -s "local://backup/db/full" --db db -t t`
3. do some DDL and DML on table
...
 "RENAME TABLE db.t to db.t1;"
 "DROP TABLE db.t1;"
 "DROP DATABASE db;"
 "CREATE DATABASE db;"
 "CREATE TABLE db.t1 (c2 CHAR(255));"
 "RENAME TABLE db.t1 to db.t;"
 "TRUNCATE TABLE db.t;"
"CREATE TABLE IF NOT EXISTS db.t_rename (c CHAR(255));"
"RENAME TABLE db.t_rename to db.t_rename2;"
"INSERT INTO db.t(c2) VALUES (‘1’);”
"INSERT INTO db.t_rename2(c) VALUES (‘1’);”
...
4. incremental backup the table
```
last_backup_ts=$(br validate decode --field="end-version" -s "local://backup/db/full" | grep -oE "^[0-9]+")
br --pd $PD_ADDR backup db -s "local://backup/db/inc" --db db --lastbackupts $last_backup_ts
```
5. restore table
`br restore table --db db --table t -s "local://backup/db/full" --pd $PD_ADDR` --ddl-batch-size=128
6. incremental restore db
`run_br restore db --db db -s "local://backup/db/inc" --pd $PD_ADDR` --ddl-batch-size=128
7. insert some data into table
```
INSERT INTO db.t(c2) VALUES ('1');"
NSERT INTO db.t_rename2(c) VALUES ('1');"
```
### 2. What did you expect to see? (Required)
after restore succeed, we can insert data into table without any other action.
### 3. What did you see instead (Required)
after restore succeed, insert table cause error `ERROR 1062 (23000): Duplicate entry '1' for key 'PRIMARY'`
### 4. What is your TiDB version? (Required)
nightly

### PR DESCRIPTION
**Pull request by fengou1**
### What problem does this PR solve?
Detail see issue

Issue Number: close #32006

Problem Summary:
Incremental batch ddl restore may cause table is not insertable, any insert sql statement will failure with ERROR info Duplicate Primary Key.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/client.go [hunk 1]
```diff
@@ -1237,15 +1238,18 @@ func (rc *Client) IsSkipCreateSQL() bool {
 // 2. create table/create and rename table: the first create table will lock down the id cache.
 // because we cannot create onExistReplace table.
 // so the final create DDL with the correct auto increment/random id won't be executed.
-func (rc *Client) DDLJobsMap() map[UniqueTableName]bool {
-	m := make(map[UniqueTableName]bool)
+func (rc *Client) GenDDLJobsMap() {
```
Comment (kennytm) [inline br/pkg/restore/client.go:1241]:
```suggestion
func (rc *Client) GenerateDDLJobtsMap() {
```

the name is too close to `GetDDLJobsMap`

**Review by kennytm — APPROVED** [review]

File: br/pkg/restore/client.go (hunk 1 above)
Comment (fengou1) [reply br/pkg/restore/client.go]: thanks, reasonable to rename it.

**Review by joccau — APPROVED** [review]
LGTM

**Pull request merged**